src/server/main_service.cc

```diff
@@ -2508,7 +2508,14 @@ GlobalState Service::SwitchState(GlobalState from, GlobalState to) {
   VLOG(1) << "Switching state from " << from << " to " << to;
   global_state_ = to;
 
-  pp_.Await([&](ProactorBase*) { ServerState::tlocal()->set_gstate(to); });
+  pp_.Await([&](ProactorBase*) {
+    ServerState::tlocal()->set_gstate(to);
+    auto* es = EngineShard::tlocal();
+    if (es && to == GlobalState::ACTIVE) {
+      DbSlice& db = namespaces->GetDefaultNamespace().GetDbSlice(es->shard_id());
+      DCHECK(db.IsFetchedItemsEmpty());
```

Comment:
Ok for this check
but why not checking the load_ref_count_ as I suggested? or better check that if cache mode flag is enabled than db slice is cash mode retuns true.
My concern is that if this logic is not correct for some reason than cache mode will be disabled